### ISSUE TITLE
Images on buttons

### ISSUE THREAD

**Issue by bvssvni** [PistonDevelopers/conrod#171]
See reddit comment http://www.reddit.com/r/rust_gamedev/comments/2mlvxa/can_conrod_buttons_have_images/

**Comment by xilec** [comment]
In my opinion, it would be awesome realize something like [WPF Content Model](https://msdn.microsoft.com/en-us/library/bb613548(v=vs.110).aspx). I understand that it is too complicated solutions for topic task. But I saw attempts to realize [Layouts](https://msdn.microsoft.com/en-us/library/ms745058(v=vs.110).aspx) in conrod and it would be greate using layouts to evade implementation of controls by hand which is just a combination of some basic controls (like Label, Button and Image for topic case).

**Comment by mitchmindtree** [comment]
Just thought I'd mention that #696 adds a primitive `Image` widget which should make it easy to add image support (or variants) for various widgets.

**Comment by pierrechevalier83** [comment]
I have been trying to tackle some rudimentary implementation of this feature:
https://github.com/PistonDevelopers/conrod/pull/783

I don't understand why if I cover a part of the button with an image, though that part seems not to be clickable anymore. It is obvious when running the all_widgets example from my pull request and clicking the button that has a texture.

Does this behaviour make sense to anyone following this? Any tip?

**Comment by pierrechevalier83** [comment]
For a quick update: I figured out what was missing. The image widget was capturing the mouse events that should have been captured by the button widget.

I know have the basic functionality working: a button with a texture. It doesn't handle positioning of the image within the button with respect to a label, though but it gives a starting point.

I also removed the part that modifies the button in the all_widgets example and wrote a dedicated button example to showcase the feature (heavily inspired from all_widgets).

This included modifying the git history, so I closed pull request #783 and opened #785.

I would welcome feedback, be it code review or suggestions for tackling more complex workflows (Like CompositeButton described in earlier comment)
